Patch-Source: https://github.com/AyatanaIndicators/libayatana-indicator/pull/80
---
From 598cc6aa58b1bbc83aefd70d6849b865e5ae195e Mon Sep 17 00:00:00 2001
From: Alessandro Astone <alessandro.astone@canonical.com>
Date: Mon, 6 Oct 2025 15:43:52 +0200
Subject: [PATCH] tests: Avoid more GDBusConnection leaks

The atk-bridge has started "leaking" a GDBusConnection since 2.57.0; disable
it since it's not needed for the test.

The GTK Wayland backend also "leaks" a GDBusConnection used to read settings
in place of xsettings; force using the X11 backend.
---
 tests/test-indicator-ng.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tests/test-indicator-ng.c b/tests/test-indicator-ng.c
index 774b0002..2b93236a 100644
--- a/tests/test-indicator-ng.c
+++ b/tests/test-indicator-ng.c
@@ -174,11 +174,13 @@ test_menu (void)
 int
 main (int argc, char **argv)
 {
-  /* gvfs, dconf, and appmenu-gtk leak GDbusConnections, which confuses
+  /* gvfs, dconf, appmenu-gtk, atk-bridge and gdk-wayland leak GDbusConnections, which confuses
    * g_test_dbus_down.  Make sure we're not using any of those.
    */
   g_setenv ("GIO_USE_VFS", "local", TRUE);
   g_setenv ("GSETTINGS_BACKEND", "memory", TRUE);
+  g_setenv ("NO_AT_BRIDGE", "1", TRUE);
+  g_setenv ("GDK_BACKEND", "x11", TRUE);
   g_unsetenv ("UBUNTU_MENUPROXY");
 
   g_test_init (&argc, &argv, NULL);
